### PR TITLE
Updates to clojure module

### PR DESCRIPTION
**Pull request by teesloane**
Hey hey!

I've been writing some more clojure in my life so I thought I'd try and add some stuff to the clojure module. This pr isn't really complete -- it works but I'm still grokking how these modules work, (although I did research into use-package as you recommended in an issue I opened ( #241  ))

Either way, feel free to review if it interests you; or to dismiss. Either way - I've got something kind of half working, and wanted to check if it would add any value to your project!

I've got the localleader keybindings working, cider is working, but I couldn't get the repl to not be a popup, as mentioned in #171 

I looked at putting a +repl solution in, but wasn't sure how to go about that. I looked at the python example:

```
;;;###autoload
(defun +python/repl ()
  "Open the Python REPL."
  (interactive)
  (process-buffer (run-python nil t t)))

```

But perhaps this would be more complicated between clojurescript and clojure with cider. I have had issue with getting repls to work with clojurescript... there seem to be some edgecases sometimes, in my experience, so I'm not sure if it's wise to bake cider into the doom-repl feature. 

One last note - I had to add `clojure` to the `init.el` to get it to work -- although that file is gitignored

Cheers!

### CODE REVIEW

File: modules/org/org/config.el [hunk 1]
```diff
@@ -6,7 +6,7 @@
   (push (file-name-directory path) load-path))
 
 ;; Custom variables
-(defvar +org-dir (expand-file-name "~/work/org/")
+(defvar +org-dir (expand-file-name "~/Dropbox/notes/")
```
Comment (teesloane) [inline modules/org/org/config.el]: woops, this should not have been commited

**Comment by hlissner** [comment]
This looks fantastic, great work! I've been away from the project for a short while, but will be back in the next few days to look into merging this asap.

**Comment by teesloane** [comment]
Sounds good! Happy to make nay changes that you need as well. Cheers!

**Comment by hlissner** [comment]
Alright! Now that I'm back I had some time to review your PR. It looks great! There are a few minor changes I'd like to ask for, to make this module more idiomatic:

1. Unnest the `(def-package! clj-refactor ...)` block and use `:after` to chain it to clojure-mode:

   ```emacs-lisp
   (def-package! clojure-mode ...)
  
   (def-package clj-refactor
     :after clojure-mode
     :config
     ...)
  
   ...
   ```

2. The `:commands (clojure-mode clojurescript-mode)` line is unnecessary. The `:mode` declarations add autoloads for these already.

3. To have shackle ignore cider, try the `:ignore` property:

   ```emacs-lisp
   (set! :popup "^\\*cider-.*" :ignore t)
   ```

4. I know little about Clojure and Cider, so I can't give any specific advice about integrating its repl into doom's repl system. If, for some reason, clojure's repl is completely incompatible with Doom's popup system, then it's best you avoid its repl system too.

    If this is not the case, doom's repl system works like this:

   + There must be an interactive function, canonically named `+module/repl`, e.g. `+python/repl`. It must open (and return) a repl buffer.
   + This function must be registered to a major mode, like so: `(set! :repl 'python-mode #'+python/repl)`
   + The repl can then be invoked from python-mode buffers using `M-x +eval/open-repl` (bound to `SPC o r` in my private module).

Hope that helps!

**Comment by teesloane** [comment]
@hlissner Thanks for the feedback!

I made the requested changes (let me know if I missed anything though!).

Regarding the repl; I think the real frustration for me was that I kept hitting `ESC` (mapped to caps on my keyboard) and that would kill the repl. After reading the shackle code I learned about noesc, so that solves my problem entirely! (before I was taking the repl, and then moving it to the right with `SPC w L` (which was a bit annoying as the cider repl boots _two_ instances of the repl when you boot a clojurescript repl.)

Should be good now, cheers!

**Comment by hlissner** [comment]
Excellent! One more thing: could you try:

```diff
-(set! :popup '("^\\*cider" :regexp t :noselect t :autokill f :autoclose f :noesc t))
+(set! :popup "^\\*cider" :regexp t :noselect t :noesc t)
```

If you omit :autokill or :autoclose, it _should_ achieve the same effect. If all works well, then it will be ready to be merged!

**Comment by teesloane** [comment]
Changes made! Looks like all still works ok!

**Comment by hlissner** [comment]
Great! And merged. Thanks again!

**Pull request merged**